Fundamentals/Code/Source-Control/index-v9.md

```diff
@@ -0,0 +1,174 @@
+
+#### Packages and Plugins
+
+The **App_Plugins** folder is the home for all third party packages installed on your site.
+
+Depending on how you installed the plugin it will affect how you choose to version control a particular third party plugin:
+
+Since plugins are installed via NuGet the installed files for individual plugins shouldn't need to be source controlled (and your deployment process should pull the packages implementation files from NuGet during the build and deployment process).
+
+:::note
+Each plugin could be different depending on its implementation and functionality. It may contain files that it would be useful to track via Source control, and also files that should be ignored: check with the plugin's supporting website/developer for more information.
+:::
+
+### What folders and files should I **include** in my source control repository?
+
+#### Front-end build
+
+A lot depends on how you maintain the front-end build of your website, e.g. are you using css preprocessors such as SCSS/LESS etc - gulp/grunt tasks to combine and minify script resources.
+But generally you will need to source control all of your website's static assets: JavaScript, Css, Fonts, Page Furniture Images etc.
+
+#### Views/Templates/Partials
+
+Umbraco site templates/views can be edited via the Umbraco Backoffice. They also reside in the `/Views` folder on disk. As these views/template often include code, it can make a lot of sense to have their changes tracked under source/version control.
+
+However this can pose a problem if the templates are updated via the backoffice outside of source control on the production environment.
+
+This is not an adviced approach since more often than not this will cause breaking changes to your website.
+You would need to manually merge these files before considering a deploy.
+Umbraco Cloud is a good solution in these scenarios, as changes via the backoffice are tracked in a Git repository automatically.
+
+#### Macros
+
+To source/version control changes to Macro implementation code, track the files in the following location:
+
+- **Partial View Macros** - stored in `/Views/MacroPartials` as .cshtml files
+
+#### Controllers/Classes/Custom Code
+
+Any supporting custom code for your application should be in version control, eg any
+- C# implementation,
+  - Surface Controllers.
+  - API Controllers.
+  - ViewModels.
+  - Helpers / Extension Methods.
+  - Services etc
+- Supporting class library projects,
+- Models generated by Modelsbuilder in SourceCodeManual or SourceCodeAuto mode
+
+...all should be added to source/version control.
+
+#### Config
+
+Your site's `apsettings.json` and `appsettings.Development.json` files contains the configuration for your Umbraco site. Add these to source control to have consistent setups, however, be careful about adding these to source control, it's important to make sure that they do not contain any secrets like API keys, and connection strings.
```

Comment:
```suggestion
### What folders and files should I **include** in my source control repository?

#### Front-end build

A lot depends on how you maintain the front-end build of your website, e.g. are you using CSS preprocessors such as SCSS/LESS etc - gulp/grunt tasks to combine and minify script resources.
But generally, you will need to source control all of your website's static assets: JavaScript, CSS, Fonts, Page Furniture Images, etc.

#### Views/Templates/Partials

Umbraco site templates/views can be edited via the Umbraco Backoffice. They also reside in the `/Views` folder on disk. As these views/templates often include code, it can make a lot of sense to have their changes tracked under source/version control.

However, this can pose a problem if the templates are updated via the backoffice outside of source control on the production environment.

This is not an advisable approach since more often than not this will cause breaking changes to your website.
You would need to manually merge these files before considering a deployment.
Umbraco Cloud is a good solution in these scenarios, as changes via the backoffice are tracked in a Git repository automatically.

#### Macros

To source/version control changes to Macro implementation code, track the files in the following location:

- **Partial View Macros** - stored in `/Views/MacroPartials` as .cshtml files

#### Controllers/Classes/Custom Code

Any supporting custom code for your application should be in version control, eg any
- C# implementation,
  - Surface Controllers.
  - API Controllers.
  - ViewModels.
  - Helpers / Extension Methods.
  - Services etc
- Supporting class library projects,
- Models generated by Modelsbuilder in SourceCodeManual or SourceCodeAuto mode

...all should be added to source/version control.

#### Config

Your site's `apsettings.json` and `appsettings.Development.json` files contain the configuration for your Umbraco site. Add these to source control to have consistent setups, however, be careful about adding these to source control, it's important to make sure that they do not contain any secrets like API keys, and connection strings.
```